episodes.js

```diff
@@ -31,1825 +31,1960 @@ function getOneEpisode() {
-      airstamp: "2019-04-29T01:00:00+00:00",
-      runtime: 90,
-      image: {
-        medium:
-          "http://static.tvmaze.com/uploads/images/medium_landscape/192/482465.jpg",
-        original:
-          "http://static.tvmaze.com/uploads/images/original_untouched/192/482465.jpg",
-      },
-      summary: "<p>Winterfell fights the Army of the Dead.</p>",
-      _links: {
-        self: {
-          href: "http://api.tvmaze.com/episodes/1623965",
-        },
-      },
-    },
-    {
-      id: 1623966,
-      url:
-        "http://www.tvmaze.com/episodes/1623966/game-of-thrones-8x04-the-last-of-the-starks",
-      name: "The Last of the Starks",
-      season: 8,
-      number: 4,
-      airdate: "2019-05-05",
-      airtime: "21:00",
-      airstamp: "2019-05-06T01:00:00+00:00",
-      runtime: 78,
-      image: {
-        medium:
-          "http://static.tvmaze.com/uploads/images/medium_landscape/195/487839.jpg",
-        original:
-          "http://static.tvmaze.com/uploads/images/original_untouched/195/487839.jpg",
-      },
-      summary:
-        "<p>The survivors plan their next steps; Cersei makes a power move.</p><p></p>",
-      _links: {
-        self: {
-          href: "http://api.tvmaze.com/episodes/1623966",
-        },
-      },
-    },
-    {
-      id: 1623967,
-      url:
-        "http://www.tvmaze.com/episodes/1623967/game-of-thrones-8x05-the-bells",
-      name: "The Bells",
-      season: 8,
-      number: 5,
-      airdate: "2019-05-12",
-      airtime: "21:00",
-      airstamp: "2019-05-13T01:00:00+00:00",
-      runtime: 79,
-      image: {
-        medium:
-          "http://static.tvmaze.com/uploads/images/medium_landscape/196/491994.jpg",
-        original:
-          "http://static.tvmaze.com/uploads/images/original_untouched/196/491994.jpg",
-      },
-      summary:
-        "<p>Varys betrays his queen, and Daenerys brings her forces to King's Landing.</p>",
-      _links: {
-        self: {
-          href: "http://api.tvmaze.com/episodes/1623967",
-        },
-      },
-    },
-    {
-      id: 1623968,
-      url:
-        "http://www.tvmaze.com/episodes/1623968/game-of-thrones-8x06-the-iron-throne",
-      name: "The Iron Throne",
-      season: 8,
-      number: 6,
-      airdate: "2019-05-19",
-      airtime: "21:00",
-      airstamp: "2019-05-20T01:00:00+00:00",
-      runtime: 80,
-      image: {
-        medium:
-          "http://static.tvmaze.com/uploads/images/medium_landscape/198/495648.jpg",
-        original:
-          "http://static.tvmaze.com/uploads/images/original_untouched/198/495648.jpg",
-      },
-      summary:
-        "<p>In the aftermath of the devastating attack on King's Landing, Daenerys must face the survivors.</p>",
-      _links: {
-        self: {
-          href: "http://api.tvmaze.com/episodes/1623968",
-        },
-      },
-    },
-  ];
+let allEpisodes = [];
+getData(showIdNumber).then((episode) => {
+  episode.forEach((element) => allEpisodes.push(element));
```

Comment:
I have a few questions about this line...

1. It looks like your variable named `episode` (singular) contains many episodes? Maybe there's a better name for that variable?
2. There's a better array method than `forEach` for producing an array from another array, one-element-per-element - do you know what it is?
3. It looks like you're copying one array into another, without modifying the elements at all. If that's the case, why not just assign `allEpisodes = something` rather than appending each element to an empty array?

Reply:
that's exactly what i need from that function: transfer response into data, unfortunately is it the best solution i found now

Reply:
Here are three different ways I think you could do what you're doing now - can you maybe rank them (and the one you already wrote), and discuss what you like / dislike about each?

1: Your current implementation:
```js
let allEpisodes = [];
getData(showIdNumber).then((episode) => {
  episode.forEach((element) => allEpisodes.push(element));
});
```

2:
```js
let allEpisodes = [];
getData(showIdNumber).then((episodes) => {
  episodes.forEach((episode) => allEpisodes.push(episode));
});
```

3:
```js
let allEpisodes = [];
getData(showIdNumber).then((episodes) => {
  allEpisodes = episodes.map(episode => episode);
});
```

4: 
```js
let allEpisodes = [];
getData(showIdNumber).then((episodes) => {
  allEpisodes = episodes;
});
```

Reply:
1 and 2 are really similar just names of variables are different and I agree that your variant is better :). 4 was still giving me a promise and I had errors in future code because of that. That's why I was trying to get fully new array with data and get rid of the promise. 3 just did not come on my mind, because push() method makes me believe the data actually gets into the array :).

Reply:
> 4 was still giving me a promise and I had errors in future code because of that.

I'm glad you tried it out! I would expect `let allEpisodes = getData(showNumber).then(...);` to have `allEpisodes` be a promise, but the code written in 4 should work ok - can you talk me through the differences between the two? :)

Reply:
:) i think, i can. The difference is that let `AllEpisodes = getData` returns a promise and 4 assigns returned promise/data to a variable. I think i did not work for me because i did not use async/await previously :) and it gave me an empty array, which i could not use further

Reply:
That's a great explanation, thanks!